### ISSUE TITLE
event.stopImmediatePropagation() cannot be unit-tested

### ISSUE THREAD

**Issue by dmitri777** [angular/angular#19216]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

Functions <DOM_ELEMENT>.dispatchEvent() and event.stopImmediatePropagation() are not working together well in Angular unit tests.

## Expected behavior
event.stopImmediatePropagation() should work in Angular unit tests exactly as in browser.

## Minimal reproduction of the problem with instructions
Create any 'hello world' / minimalistic Angular project with Angular CLI, and add the following trivial unit test in the file with existing (auto-generated) sample unit tests:

```
let count = 0;
it('should test listeners', () => {
	document.body.addEventListener('click', (event: MouseEvent) => {
		event.stopImmediatePropagation();
		console.log('listener 1 running');
		count++;
		event.stopImmediatePropagation();
	});

	document.body.addEventListener('click', (event: MouseEvent) => {
		console.log('listener 2 running');
		count++;
	});

	document.body.addEventListener('click', (event: MouseEvent) => {
		console.log('listener 3 running');
		count++;
	});

	document.body.dispatchEvent(new MouseEvent('click', { bubbles: true }));

	expect(count).toEqual(1); // fails - is actually 3, even with event.stopImmediatePropagation() in the 1st listener
});
```

Run unit tests suit (ng test). The unit test added by us fails, because all 3 handlers do run, even with event.stopImmediatePropagation() in the 1st listener, which is supposed to stop two other handers from executing. 

This unit test works perfectly in Browser (using Jasmine stand-alone setup), only first handler is executed. We need the same behaviour in Angular unit tests.

## What is the motivation / use case for changing the behavior?
It seems to be a bug, preventing proper testing of events in Angular.

## Environment

<pre><code>
Angular version: 4.3.6

Browser:
- [x] Chrome (desktop) version 60.0.3112.113 (Official Build) (64-bit)
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: v8.4.0
- Platform: MacOS Sierra 10.12.6

Others:

</code></pre>

**Comment by JiaLiPassion** [comment]
@dmitri777 , I have submit a PR https://github.com/angular/angular/pull/19222 to fix this one.

**Comment by pkozlowski-opensource** [comment]
Sounds like it was fixed via #19222
